modules/core/src/main/java/org/apache/ignite/dump/DumpReader.java

```diff
@@ -108,11 +109,22 @@ public DumpReader(DumpReaderConfiguration cfg, IgniteLogger log) {
 
                 AtomicBoolean skip = new AtomicBoolean(false);
 
+                Map<Integer, Set<Integer>> groups = cfg.skipCopies() ? new HashMap<>() : null;
+
                 for (Map.Entry<Integer, List<String>> e : grpToNodes.entrySet()) {
                     int grp = e.getKey();
```

Comment:
YOu can init group set here and omit check on each part:

```
groups.computeIfAbsent(grp, key -> new HashSet<>())
```
```
                            if (groups != null && !groups.get(grp).add(part)) {

Reply:
Changed.